datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -0,0 +1,248 @@
+        default_fields['skip_proxy'] = init_config.get('skip_proxy', default_fields['skip_proxy'])
+
+        # Populate with the default values
+        config = {field: instance.get(field, value) for field, value in iteritems(default_fields)}
+
+        # Support non-standard (usually legacy) configurations, for example:
+        # {
+        #     'disable_ssl_validation': {
+        #         'name': 'ssl_verify',
+        #         'default': False,
+        #         'invert': True,
+        #     },
+        #     ...
+        # }
+        if remapper is None:
+            remapper = {}
+
+        remapper.update(DEFAULT_REMAPPED_FIELDS)
+
+        for remapped_field, data in iteritems(remapper):
+            field = data.get('name')
+
+            # Ignore fields we don't recognize
+            if field not in STANDARD_FIELDS:
+                continue
+
+            # Ignore remapped fields if the standard one is already used
+            if field in instance:
+                continue
+
+            # Get value, with a possible default
+            value = instance.get(remapped_field, data.get('default', default_fields[field]))
+
+            # Invert booleans if need be
+            if data.get('invert'):
+                value = not is_affirmative(value)
+
+            config[field] = value
+
+        # http://docs.python-requests.org/en/master/user/advanced/#timeouts
+        timeout = int(config['timeout'])
+
+        # http://docs.python-requests.org/en/master/user/quickstart/#custom-headers
+        # http://docs.python-requests.org/en/master/user/advanced/#header-ordering
+        headers = None
+        if config['headers']:
+            headers = OrderedDict((key, str(value)) for key, value in iteritems(config['headers']))
+
+        # http://docs.python-requests.org/en/master/user/authentication/
+        auth = None
+        if config['username'] and config['password']:
+            auth = (config['username'], config['password'])
+
+        # http://docs.python-requests.org/en/master/user/advanced/#ssl-cert-verification
+        verify = True
+        if isinstance(config['ssl_ca_cert'], string_types):
+            verify = config['ssl_ca_cert']
+        elif not is_affirmative(config['ssl_verify']):
+            verify = False
+
+        # http://docs.python-requests.org/en/master/user/advanced/#client-side-certificates
+        cert = None
+        if isinstance(config['ssl_cert'], string_types):
+            if isinstance(config['ssl_private_key'], string_types):
+                cert = (config['ssl_cert'], config['ssl_private_key'])
+            else:
+                cert = config['ssl_cert']
+
+        # http://docs.python-requests.org/en/master/user/advanced/#proxies
+        no_proxy_uris = None
+        if is_affirmative(config['skip_proxy']):
+            proxies = PROXY_SETTINGS_DISABLED.copy()
+        else:
+            proxies = config['proxy'] or init_config.get('proxy')
```

Comment:
Let's add a comment to describe the order of precedence here `agent config < init config < instance config`